lib/utils.ts

```diff
@@ -159,11 +159,15 @@ export function existsInArray<T>(inArray: T[], value: T) {
   return inArray.indexOf(value) >= 0;
 }
 
-/**
- * Removes an object from the array by reference equality or shallow value equality
+ * Removes an object from the array by strict equality
  */
 export function removeFromArray<T>(array: T[], value: T) {
-  array.splice(array.indexOf(value), 1);
+  const valueIndex = array.indexOf(value);
+  if (valueIndex === -1) {
```

Comment:
Hmm..this is more a matter of design opinion, but the warn here seems odd. We should just instead describe the expected behavior in the method comment. "If the value does not exist in the array, no action is taken." In my experience you either throw an exception here (Python) or do a no-op (most other languages). The in between of a warn where a human has to be paying attention to catch it seems too arbitrary. What do you think?

Reply:
Yes, I wasn't very enthused by this solution either but I lacked the domain knowledge to make another decision at the time. I've seen warnings pop up in the console while I was running tests, which made me see that this situation could sometimes present itself. What I'm wondering is how much of it is intentional… Maybe this calls for two different functions, a strict one for when you know the element to be in the array and a soft one for when you don't?

Reply:
Warnings in the console written by me tend to follow one of these patterns:

- You're accessing a deprecated function
- You're using a function incorrectly but not in a way that would jeopardize behavior

But in this case as long as we document that it is a no-op if the element doesn't exist, then it is not being used incorrectly. I think it's safe to go without this warning in that case.

Reply:
I pushed some new changes to align with your recommendation, I'll let you resolve the conversation if you think it's good now :)